src/charm.py

```diff
@@ -1403,7 +1416,6 @@ def _on_leader_elected(self, event: LeaderElectedEvent) -> None:  # noqa: C901
 
         if not self.get_secret(APP_SCOPE, "internal-ca"):
             self.tls.generate_internal_peer_ca()
-            self.tls.generate_internal_peer_cert()
```

Comment:
If the internal cert generates before the IP is set to the peer data, the hostname will be used as common name. It also causes issues on Juju 4.